pkg/output/output.go

```diff
@@ -0,0 +1,79 @@
+package output
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"path"
+)
+
+// Closer is used to close a file when done
+type Closer func() error
+
+// Outputter is an interface for any object which can write the processed output.
+type Outputter interface {
+	// Writer returns a io.StringWriter which can used to write to the output. The caller must ensure that
+	// the parent directory already exists and the close function is called after the caller is done.
+	Writer(parent string, name string, permissions os.FileMode) (Closer, io.StringWriter, error)
+	// MakeDir creates a directory. The caller must ensure that the parent already exists.
```

Comment:
do we need to have an extra call for mkdir? when we attempt to write a file, can we not just `os.MkdirAll`?

Reply:
I'm trying to preserve the permissions of the directories and this requires that we don't use `MkdirAll`.

Reply:
what kind of special permissions do you expect subfolders to have in a must-gather?

Reply:
Also I'm trying to avoid the situation where obfuscation causes directories to be overwritten or merged.

Reply:
I would just say that we don't ever change the directory structure (aside from omissions) and that means that we do not try to obfuscate it. I'm 99% certain that there won't be anything to scrub in the directory names anyway